### PR TITLE
[MOBILE-2714] Update Xamarin to iOS 15.0.1

### PR DESCRIPTION
**Pull request by khmMouna**
### What do these changes do?
Update xamarin to latest iOS Airship SDK.

### How did you verify these changes?
Run sample app

PS: Not ready yet for review.

### CODE REVIEW

File: CHANGELOG.md [hunk 1]
```diff
@@ -1,5 +1,14 @@
 # Airship Xamarin Changelog
 
+## Version 15.1.0 - Dec 2, 2021
```
Comment (rlepinski) [inline CHANGELOG.md:3]: This should be 15.0.1

**Comment by rlepinski** [comment]
WE need to add block action bindings - https://github.com/urbanairship/urbanairship-xamarin/issues/187 

to make custom actions easier.

File: src/AirshipBindings.iOS.Core/ApiDefinitions.cs [hunk 2]
```diff
@@ -260,6 +260,12 @@ interface UAActivityViewController : IUIPopoverControllerDelegate, IUIPopoverPre
 		IntPtr Constructor(NSObject[] activityItems, [NullAllowed] UIActivity[] applicationActivities);
 	}
 
+    // @interface UABlockAction : NSObject <UAAction>
+    [BaseType(typeof(NSObject))]
+    interface UABlockAction : IUAAction
```
Comment (rlepinski) [inline src/AirshipBindings.iOS.Core/ApiDefinitions.cs:265]: Not sure if this is enough for it to  work, i think you need a constructor as well

Comment (rlepinski) [reply src/AirshipBindings.iOS.Core/ApiDefinitions.cs:265]: You should double check you can use it like a normal block action

**Comment by rlepinski** [comment]
@khmMouna I think that looks right, did you try using it in a sample? What is the sample code for that?

**Review by rlepinski — APPROVED** [review]

**Review by oristanovic — APPROVED** [review]

**Comment by khmMouna** [comment]
@rlepinski it works fine now: 
Here is the code to use it:
```

            UABlockAction customAction = new UABlockAction((args, completionHandler) =>
            {
                Console.WriteLine("UABlockAction Value {0}", args.Value);
                Console.WriteLine("UABlockAction  Situation {0}", args.Situation);
                completionHandler(new UAActionResult());
            });
            UAirship.Shared.ActionRegistry.RegisterAction(customAction, "action_name");

            NSObject value = FromObject("action_value");
            UAActionRunner.RunAction(customAction, value, UASituation.ManualInvocation);
```

File: CHANGELOG.md [hunk 3]
```diff
@@ -1,13 +1,15 @@
...
 Minor release updating to latest iOS Airship SDK.
 
 ### Changes
-- Updated iOS SDK to 16.1.1
+- Updated iOS SDK to 16.1.2
 - Fixed Message Center navigation style on iOS 15
 - Fixed running actions from a notification action button
 - Fixed channel registration causing extra attribute operations in the RTDS stream
+- Add BlockAction type.
+- Fix packages references conflits in the preference center nuspec.
```
Comment (oristanovic) [inline CHANGELOG.md:12]: conflicts*

**Review by oristanovic — APPROVED** [review]

**Pull request merged**